nmproxy/stun/stun.go

```diff
@@ -81,16 +113,18 @@ func GetHostInfo(stunList []nmmodels.StunServer, proxyPort int) (info models.Hos
 }
 
 // compare ports and endpoints between stun results to determine nat type
-func getNatType(endpointList []stun.XORMappedAddress, proxyPort int) string {
-	natType := config.DOUBLE_NAT
+func getNatType(endpointList []stun.XORMappedAddress, currentPublicIP string, stunPort int) string {
+	natType := nmmodels.NAT_Types.Double
 	ip1 := endpointList[0].IP
 	ip2 := endpointList[1].IP
 	port1 := endpointList[0].Port
 	port2 := endpointList[1].Port
-	if ip1.Equal(ip2) && port1 == port2 && port1 == proxyPort {
-		natType = config.SYMMETRIC_NAT
-	} else if ip1.Equal(ip2) && port1 == port2 && port1 != proxyPort {
-		natType = config.ASYMMETRIC_NAT
+	if ip1.Equal(ip2) && IsPublicIP(ip1) && DoesIPExistLocally(ip1) {
+		natType = nmmodels.NAT_Types.Public
```

Comment:
why is this `DoesIPExistLocally` check required?

Reply:
Because the ip may still be classified as a public IP, but does not reflect an interface's assigned address